core/src/test/java/com/epam/esm/service/impl/CertificateServiceImplTest.java

```diff
@@ -0,0 +1,240 @@
+package com.epam.esm.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.epam.esm.TestConfig;
+import com.epam.esm.TestEntityProvider;
+import com.epam.esm.dao.CertificateDao;
+import com.epam.esm.dao.TagDao;
+import com.epam.esm.dao.entity.Certificate;
+import com.epam.esm.dao.entity.Tag;
+import com.epam.esm.dto.CertificateDto;
+import com.epam.esm.exception.AbstractLocalizedCustomException;
+import com.epam.esm.exception.InappropriateBodyContentException;
+import com.epam.esm.exception.ResourceNotFoundException;
+import com.epam.esm.mapper.impl.CertificateMapperImpl;
+import com.epam.esm.service.validation.InputDataValidator;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@ExtendWith({SpringExtension.class, MockitoExtension.class})
+@ContextConfiguration(classes = {TestConfig.class})
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+class CertificateServiceImplTest {
+
+	@Mock
+	private CertificateDao certificateDao;
+
+	@Mock
+	private TagDao tagDao;
+
+	@Mock
+	CertificateMapperImpl certificateMapper;
+
+	@Mock
+	private InputDataValidator validator;
+
+	@InjectMocks
+	private CertificateServiceImpl certificateService;
+
+	@Autowired
+	TestEntityProvider entityProvider;
+
+	@BeforeEach
+	public void setUp() {
+		MockitoAnnotations.openMocks(this);
+	}
+
+	@Test
+	void findByIdShouldReturnCertificateDto() {
+		CertificateDto certificateDtoExpected = entityProvider.getCertificate3dto();
+		long certificateDtoId = certificateDtoExpected.getId();
+		Certificate certificate = entityProvider.getCertificate3();
+		when(certificateDao.readById(certificateDtoId)).thenReturn(Optional.of(certificate));
+		when(certificateMapper.convertToDto(certificate)).thenReturn(certificateDtoExpected);
+		assertEquals(certificateDtoExpected, certificateService.findById(certificateDtoId));
+		verify(certificateDao, times(1)).readById(certificateDtoId);
+		verify(certificateMapper).convertToDto(certificate);
```

Comment:
add verifyNoMoreInteractions for all mocks in all methods

Reply:
done